praw/models/reddit/subreddit.py

```diff
@@ -2155,3 +2174,103 @@ def revisions(self, **generator_kwargs):
         url = API_PATH['wiki_revisions'].format(subreddit=self.subreddit)
         return WikiPage._revision_generator(
             self.subreddit, url, generator_kwargs)
+
+
+class SubredditEmoji(object):
+    """Provides a set of functions to a Subreddit for emoji."""
+
+    def __getitem__(self, name):
+        """Lazily return the Emoji for the subreddit named ``name``.
+
+        This method is to be used to fetch a specific emoji url, like so:
+
+        .. code:: python
+
+           emoji = reddit.subreddit('iama').emoji['cake']
+           print(emoji)
+
+        """
+        if e.name == name for e in self.subreddit.emoji:
+            return e for e in self.subreddit.emoji if e.name == name
+        else:
+            return None
+
+    def __init__(self, subreddit):
+        """Create a SubredditEmoji instance.
+
+        :param subreddit: The subreddit whose emoji are affected.
+        :param reddit: The reddit instance.
+
+        """
+        self.subreddit = subreddit
+        self.reddit = self.subreddit._reddit
+
+    def __iter__(self):
+        """Iterate through the Emoji for the subreddit.
+
+        This method is to be used to discover all emoji for a subreddit:
+
+        .. code:: python
+
+           for emoji in reddit.subreddit('praw_test').emoji:
+               print(emoji)
+
+        """
+        response = self.subreddit._reddit.get(
+            API_PATH['emoji_list'].format(subreddit=self.subreddit))
```

Comment:
I really like what you're thinking here @jarhill0. For this round, let's not worry about caching. Otherwise this PR will continue to keep getting larger and larger. We can improve upon things once they are in to master.